lib/solr/response.rb

```diff
@@ -1,47 +1,49 @@
-require 'solr/response/facet_value'
-require 'solr/response/field_facets'
+require 'solr/response/header'
+require 'solr/response/http_status'
+require 'solr/response/solr_error'
 require 'solr/response/parser'
-require 'solr/response/spellcheck'
 
 module Solr
   class Response
-    attr_reader :documents, :available_facets, :spellcheck
+    def self.from_raw_response(response)
+      Solr::Response::Parser.new(response).parse
+    end
 
-    class << self
-      def empty
-        new(documents: Solr::DocumentCollection.empty)
-      end
+    attr_reader :header, :http_status, :solr_error
 
-      def empty_grouped
-        new(documents: Solr::GroupedDocumentCollection.empty)
-      end
+    def initialize(header:, http_status: HttpStatus.ok, solr_error: SolrError.none)
+      @header = header
+      @http_status = http_status
+      @solr_error = solr_error
+      freeze
+    end
 
-      def manual_grouped_listing_documents(listing_ids)
-        documents = listing_ids.map { |id| Solr::Document.new(id: id, model_name: 'Listing') }
-        group_counts = listing_ids.reduce({}) do |acc, id|
-          acc[id] = 1
-          acc
-        end
-        new(documents: Solr::GroupedDocumentCollection.new(
-          documents: documents,
-          total_count: listing_ids.count,
-          group_counts: group_counts
-        ))
-      end
+    def ok?
+      header.ok?
+    end
+
+    def error?
+      !ok?
     end
 
-    def initialize(documents:, available_facets: [], spellcheck: Solr::Response::Spellcheck.empty)
-      @documents = documents
-      @available_facets = available_facets
-      @spellcheck = spellcheck
+    def status
+      if header.status.zero?
+        'OK'
```

Comment:
What do you think about moving this `OK` string to a constant? I see that it's being used in more places.

Reply:
yeah